plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -391,9 +449,72 @@ func (c *CiscoTelemetryMDT) parseKeyField(tags map[string]string, field *telemet
 	}
 }
 
+func (c *CiscoTelemetryMDT) parseRib(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+	// RIB
+	measurement := path
+	for _, subfield := range field.Fields {
+		//For Every table fill the keys which are vrfName, address and masklen
+		switch subfield.Name {
+		case "vrfName", "address", "maskLen":
+			tags[subfield.Name] = decodeTag(subfield)
+		}
+		if value := decodeValue(subfield); value != nil {
+			grouper.Add(measurement, tags, timestamp, subfield.Name, value)
+		}
+		if subfield.Name != "nextHop" {
+			continue
+		}
+		//For next hop table fill the keys in the tag - which is address and vrfname
+		for _, subf := range subfield.Fields {
+			for _, ff := range subf.Fields {
+				switch ff.Name {
+				case "address", "vrfName":
+					key := "nextHop/" + ff.Name
+					tags[key] = decodeTag(ff)
+				}
+				if value := decodeValue(ff); value != nil {
+					name := "nextHop/" + ff.Name
+					grouper.Add(measurement, tags, timestamp, name, value)
+				}
+			}
+		}
+	}
+}
+
+func (c *CiscoTelemetryMDT) parseClassAttributeField(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+	// DME structure: https://developer.cisco.com/site/nxapi-dme-model-reference-api/
+	var nxAttributes *telemetry.TelemetryField
+	isDme := strings.Contains(path, "sys/")
+	if path == "rib" {
+		//handle native data path rib
+		c.parseRib(grouper, field, prefix, path, tags, timestamp)
+		return
+	}
+	if field == nil || !isDme {
+		return
+	}
+
+	if len(field.Fields) >= 1 && field.Fields[0] != nil && field.Fields[0].Fields != nil && field.Fields[0].Fields[0] != nil && field.Fields[0].Fields[0].Fields[0].Name != "attributes" {
+		return
+	}
+	nxAttributes = field.Fields[0].Fields[0].Fields[0].Fields[0]
```

Comment:
This logic seems to have a bunch of bad paths through it. eg:
- if `len(field.Fields)` is 0, line 500 will panic
- if len(field.Fields[0].Fields) is 0, 500 will panic.
etc.
try exiting on all the cases where len() is 0. nil checks aren't necessary on slices because len() of a nil slice returns 0.

Reply:
You are right and I think the check on line 497 is the opposite of what it should be, isn't it?

Reply:
Looks like the updated checks address the concern of Fields being empty or nil, marking this conversation as resolved.